cobbler/actions/buildiso/netboot.py

```diff
@@ -431,7 +431,15 @@ def filter_systems(self, selected_items: List[str] = None) -> list:
         """
         if selected_items is None:
             selected_items = []
-        return self.filter_items(self.api.systems(), selected_items)
+        found_systems = self.filter_items(self.api.systems(), selected_items)
+        # Now filter all systems out that are image based as we don't know about their kernel and initrds
+        return_systems = []
+        for system in found_systems:
+            # All systems not underneath a profile should be skipped
+            if system.get_conceptual_parent().TYPE_NAME == "profile":
+                return_systems.append(system)
+        # Now finally return
+        return return_systems
```

Comment:
nitpick: this could be a list comprehension
```suggestion    
        # All systems not underneath a profile should be skipped
        return [s for s in found_system if s.get_conceptual_parent().TYPE_NAME == "profile"]
```

Reply:
I always find list comprehensions hard to read. I will stick this time with my explicit version. Thanks for the suggestion still!